test/integration/hat-threads-run-every-frame.js

```diff
@@ -103,6 +104,100 @@ test('edge activated hat thread not added twice', t => {
     });
 });
 
+
+/**
+ * Duplicating a sprite should also track duplicated edge activated hat in
+ * runtime's _edgeActivatedHatValues map.
+ */
+test('edge activated hat should trigger for both sprites when sprite is duplicated', t => {
+
+    // Project that is similar to loudness-hat-block.sb2, but has code on the sprite so that
+    // the sprite can be duplicated
+    const projectWithSpriteUri = path.resolve(__dirname, '../fixtures/edge-triggered-hat.sb3');
+    const projectWithSprite = readFileToBuffer(projectWithSpriteUri);
+
+    const vm = new VirtualMachine();
+    vm.attachStorage(makeTestStorage());
+
+    // Start VM, load project, and run
+    t.doesNotThrow(() => {
+        // Note: don't run vm.start(), we handle calling _step() manually in this test
+        vm.runtime.currentStepTime = 0;
+        vm.clear();
+        vm.setCompatibilityMode(false);
+        vm.setTurboMode(false);
+
+        vm.loadProject(projectWithSprite).then(() => {
+            t.equal(vm.runtime.threads.length, 0);
+
+            vm.runtime._step();
+            t.equal(vm.runtime.threads.length, 1);
+            checkIsHatThread(t, vm, vm.runtime.threads[0]);
+            t.assert(vm.runtime.threads[0].status === Thread.STATUS_RUNNING);
+            // Run execute on the thread to populate the runtime's
+            // _edgeActivatedHatValues object
+            execute(vm.runtime.sequencer, vm.runtime.threads[0]);
+            t.equal(Object.keys(vm.runtime._edgeActivatedHatValues).length, 1);
```

Comment:
I wonder if there should be a formal (internal?) API for asking about a edge hat value. (Probably on target since they are target specific now?)

Reply:
Also maybe it should test that the hat value is specifically `undefined` or some other specific other?

Reply:
Mostly I just want to check about their separate existences.. after duplicating / cloning. There are other tests that make sure edge hats work properly